### PR TITLE
test: add test_pvc_storage_class_name_from_backup_volume

### PR DESCRIPTION
**Pull request by chriscchien**
Add test case `test_pvc_storage_class_name_from_backup_volume`

ref: [7454](https://github.com/longhorn/longhorn/issues/7454)

test [result](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5785/)

### CODE REVIEW

File: manager/integration/tests/test_basic.py [hunk 1]
```diff
@@ -1891,17 +1892,89 @@ def test_pvc_storage_class_name_from_backup_volume(): # NOQA
...
+    wait_for_backup_completion(client, volume_name, snapshot.name)
+
+    # in nfs backupstore, bv.storageClassName sometimes were empty
+    # due to timing issue
+    for i in range(RETRY_COMMAND_COUNT):
+        bv, b = find_backup(client, volume_name, snapshot.name)
+        if bv.storageClassName != "":
+            break
+        time.sleep(RETRY_INTERVAL)
+    assert bv.storageClassName == storage_class['metadata']['name']
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_basic.py:1942]: Just to confirm, according to the test skeleton `status.storageClassName` is asserted. Do we have two fields `StorageClassName` and `status.storageClassName`?

File: manager/integration/tests/test_basic.py [hunk 2]
```diff
@@ -1891,17 +1892,89 @@ def test_pvc_storage_class_name_from_backup_volume(): # NOQA
...
+    create_pv_for_volume(client, core_api, volume, restore_name)
+    create_pvc_for_volume(client, core_api, volume, restore_name)
+
+    claim = core_api.\
+        read_namespaced_persistent_volume_claim(name=restore_name,
+                                                namespace='default')
+
+    assert claim.spec.storage_class_name == storage_class['metadata']['name']
+
+    backup_pod = pod_make(name="backup-pod")
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_basic.py:1965]: As we are using same fixture twice, can you please check that after the test completion, all the pods are teared off?

**Review by khushboo-rancher — COMMENTED** [review]
In general LGTM, couple of doubts

File: manager/integration/tests/test_basic.py (hunk 1 above)
Comment (chriscchien) [reply manager/integration/tests/test_basic.py:1942]: Checked and there are only `status.storageClassName` in `backupvolumes.longhorn.io`
```bash
> k get backupvolume pvc-d8869f46-c205-40c3-851e-85c8f37c1410 -n longhorn-system -o yaml | grep storageClassName
  storageClassName: longhorn-test
```

File: manager/integration/tests/test_basic.py (hunk 2 above)
Comment (chriscchien) [reply manager/integration/tests/test_basic.py:1965]: Hi @khushboo-rancher , yes the pods and other resources in this test case all teared off  after test completion (using  longhorn-test maser to examine.)

**Review by khushboo-rancher — APPROVED** [review]

**Comment by chriscchien** [comment]
@mergify backport v1.5.x

**Pull request merged**